pulp_docker/app/views.py

```diff
@@ -135,26 +139,46 @@ def get(self, request, path, tag_name):
         Return a response to the "GET" action.
         """
         distribution = get_object_or_404(DockerDistribution, base_path=path)
-        try:
-            ca = ContentArtifact.objects.get(
-                content__in=distribution.publication.repository_version.content,
-                relative_path=tag_name)
-            content = ca.content.cast()
-            if content.manifest:
-                headers = {'Content-Type': content.manifest.media_type}
+
+        if MEDIA_TYPE.MANIFEST_LIST in request.META['HTTP_ACCEPT']:
+            try:
+                tag = ManifestListTag.objects.get(
+                    pk__in=distribution.publication.repository_version.content,
+                    name=tag_name
+                )
+            # If there is no manifest list tag, try again with manifest tag.
+            except ObjectDoesNotExist:
+                pass
             else:
-                headers = {'Content-Type': content.manifest_list.media_type}
-        except ObjectDoesNotExist:
-            pass
-        else:
-            artifact = ca.artifact
-            if artifact:
-                return self._dispatch(artifact.file.name, headers)
+                headers = {'Content-Type': MEDIA_TYPE.MANIFEST_LIST}
+                return self._dispatch_tag(tag, path, headers)
+
+        if MEDIA_TYPE.MANIFEST_V2 in request.META['HTTP_ACCEPT']:
+            try:
+                tag = ManifestTag.objects.get(
```

Comment:
@dkliban, this is where we try again. the docker client, if it contains the MANIFEST_LIST Content-Type will also have the MANIFEST_V2 Content-Type

Reply:
I didn't realize that both content types are sent in the Accept header. This code makes sense.